test-network-function/lifecycle/suite.go

```diff
@@ -425,7 +430,9 @@ func uncordonNode(node string) {
 func testPodAntiAffinity(env *config.TestEnvironment) {
 	var deployments dp.DeploymentMap
 	ginkgo.When("CNF is designed in high availability mode ", func() {
-		ginkgo.It("Should set pod replica number greater than 1 and corresponding pod anti-affinity rules in deployment", func() {
+		testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHighAvailabilityBestPractices)
+		ginkgo.It(testID, func() {
+			ginkgo.By("Should set pod replica number greater than 1 and corresponding pod anti-affinity rules in deployment")
```

Comment:
"By" doesn't really show. Same as my comment on the other PR, I don't see why the original text has to be removed. If the new addition has to be at the end, maybe add a prefix so it's sort of separated. How about:

ginkgo.It(fmt.Sprintf("Should set pod replica number greater than 1 and corresponding pod anti-affinity rules in deployment. Test Id: %s", identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHighAvailabilityBestPractices)), func(){

Reply:
Or maybe put it into when, but at the front? That way we don't need to repeat suite name.

Reply:
this is used in xml file only. is there really a need to write this very long description in the test?
we have that in identifiers. By  will only show up if you have verbose.
in the script run-cnf-test; we need to pass -gingko.v

Reply:
Seems like the By text will only appear if the test fails. As per ginkgo [doc](https://onsi.github.io/ginkgo/#documenting-complex-its-by):
`The string passed to By is emitted via the GinkgoWriter. If a test succeeds you won’t see any output beyond Ginkgo’s green dot. If a test fails, however, you will see each step printed out up to the step immediately preceding the failure. Running with ginkgo -v always emits all steps.`

Reply:
@jun-rh  is on PTO.
@edcdavid @greyerof do you think I can leave it as-is and pass the -ginko.v all the times ?
or I should update the test identifier to include the description as well ?

Reply:
I'd do both things:
1. Use jun's It text version (first jun's comment). I like ginkgo's/bdd descriptive text for the It blocks.
2. pass always the -ginkgo.v.  This could create large logs, but can be disabled at any moment.

Reply:
one of the main objective for this change is to make it easier to identify tests and allow users to skip tests.
here's a log when running the program
```
./run-cnf-suites.sh -f lifecycle
...
...
lifecycle 
  lifecycle-scaling
  /Users/shamadi/redhat/go/test-network-function/test-network-function/lifecycle/suite.go:169
INFO[0005] Testing deployment=test, namespace=tnf pod name=test-697ff58f87-hqxdm 
2021/08/20 14:23:50 Sent: "oc -n tnf get deployments -o custom-columns=NAME:.metadata.name,REPLICAS:.spec.replicas,READY:.status.readyReplicas,UPDATED:.status.updatedReplicas,AVAILABLE:.status.availableReplicas,UNAVAILABLE:.status.unavailableReplicas && echo END_OF_TEST_SENTINEL\n"
2021/08/20 14:23:50 Match for RE: "(?s).+((.|\n)*END_OF_TEST_SENTINEL\n)" found: ["NAME      REPLICAS   READY   UPDATED   AVAILABLE   UNAVAILABLE\npartner   1          1       1         1           <none>\ntest      2          2       2         2           <none>\nEND_OF_TEST_SENTINEL\n" "END_OF_TEST_SENTINEL\n" ""] Buffer: "NAME      REPLICAS   READY   UPDATED   AVAILABLE   UNAVAILABLE\npartner   1          1       1         1           <none>\ntest      2          2       2         2           <none>\nEND_OF
```
now, the user can easily detect the suite and the test name. 
then he can use the command 
```
./run-cnf-suites.sh -f lifecycle -s lifecycle-scaling
``` 
the test description is not unique and can mess up the regular expressions.
If we want to add it to When; that should be fine, the only drawback is that it will appear on JUNIT
```
      </testcase>
      <testcase name="lifecycle lifecycle-pod-scheduling" classname="CNF Certification Test Suite" time="2.901942578"></testcase>
      <testcase name="lifecycle when Test terminationGracePeriod  lifecycle-pod-termination-grace-period" classname="CNF Certification Test Suite" time="0.202937519"></testcase>
      <testcase name="lifecycle when Testing PUTs are configured with pre-stop lifecycle lifecycle-container-shutdown" classname="CNF Certification Test Suite" time="0.220083457"></testcase>
      <testcase name="lifecycle when CNF is designed in high availability mode  lifecycle-pod-high-availability" classname="CNF Certification Test Suite" time="0.421853337"></testcase>
      <testcase name="lifecycle lifecycle-pod-recreation" classname="CNF Certification Test Suite" time="0">
          <skipped></skipped>
      </testcase>
```
see the difference between lifecycle-pod-scheduling and the others...

if something went bad, then we can rely on the logs to give us more data

Reply:
test name list  .e.g.  lifecycle-scaling etc,  should be added into README.md, so user can pick name from list to -s to cancel test